cdap-proto/src/main/java/co/cask/cdap/proto/security/CheckAuthorizedRequest.java

```diff
@@ -16,23 +16,21 @@
 
 package co.cask.cdap.proto.security;
 
+import co.cask.cdap.api.annotation.Beta;
 import co.cask.cdap.proto.id.EntityId;
 import com.google.common.base.Preconditions;
 
 import java.util.Set;
 
 /**
- * Request for checking if a user can perform certain actions on an entity.
+ * Request for checking if a principal can perform certain actions on an entity.
  */
+@Beta
 public class CheckAuthorizedRequest extends AuthorizationRequest {
 
-  public CheckAuthorizedRequest(EntityId entity, String user, Set<Action> actions) {
-    super(entity, user, actions);
-    Preconditions.checkArgument(user != null, "user is required");
+  public CheckAuthorizedRequest(EntityId entity, Principal principal, Set<Action> actions) {
+    super(entity, principal, actions);
+    Preconditions.checkArgument(principal != null, "principal is required");
```

Comment:
But isn't it basically null and not an illegal argument ? 
But if you want to keep it its fine with me.